packages/account-tree-controller/src/backup-and-sync/user-storage/network-operations.ts

```diff
@@ -0,0 +1,277 @@
+import { SDK } from '@metamask/profile-sync-controller';
+
+import {
+  USER_STORAGE_GROUPS_FEATURE_KEY,
+  USER_STORAGE_WALLETS_FEATURE_ENTRY_KEY,
+  USER_STORAGE_WALLETS_FEATURE_KEY,
+} from './constants';
+import {
+  formatWalletForUserStorageUsage,
+  formatGroupForUserStorageUsage,
+  parseWalletFromUserStorageResponse,
+  parseGroupFromUserStorageResponse,
+  parseLegacyAccountFromUserStorageResponse,
+} from './format-utils';
+import { executeWithRetry } from './network-utils';
+import type { AccountGroupMultichainAccountObject } from '../../group';
+import type { AccountWalletEntropyObject } from '../../wallet';
+import type {
+  BackupAndSyncContext,
+  LegacyUserStorageSyncedAccount,
+  UserStorageSyncedWallet,
+  UserStorageSyncedWalletGroup,
+} from '../types';
+
+/**
+ * Retrieves the wallet from user storage.
+ *
+ * @param context - The backup and sync context.
+ * @param entropySourceId - The entropy source ID.
+ * @returns The wallet from user storage or null if not found or invalid.
+ */
```

Comment:
Technically, all those functions (from this file) calling `executeWithRetry` can throw, so we could add a `@throw` entry maybe?